pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -110,58 +157,68 @@ func (r *NetworkInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request)
-		message := fmt.Sprintf("Failed to read network config %s when creating NSX VPC", ncName)
-		log.Info(message)
-		updateFail(r, ctx, networkInfoCR, &err, r.Client, nil)
-		return common.ResultRequeueAfter10sec, errors.New(message)
-	}
+
+	ncName := nc.Name
 	log.Info("Fetched network config from store", "NetworkConfig", ncName)
-	vpcNetworkConfiguration := &v1alpha1.VPCNetworkConfiguration{}
-	err = r.Client.Get(ctx, types.NamespacedName{Name: commonservice.SystemVPCNetworkConfigurationName}, vpcNetworkConfiguration)
+
+	systemVpcNetCfg := &v1alpha1.VPCNetworkConfiguration{}
+	err = r.Client.Get(ctx, types.NamespacedName{Name: commonservice.SystemVPCNetworkConfigurationName}, systemVpcNetCfg)
 	if err != nil {
 		log.Error(err, "Failed to get system VPCNetworkConfiguration")
 		updateFail(r, ctx, networkInfoCR, &err, r.Client, nil)
-		return common.ResultRequeueAfter10sec, err
-	}
-	gatewayConnectionReady, _, err := getGatewayConnectionStatus(ctx, vpcNetworkConfiguration)
-	if err != nil {
-		log.Error(err, "Failed to get the gateway connection status", "NetworkInfo", req.NamespacedName)
+		setNSNetworkReadyCondition(ctx, r.Client, req.Namespace, nsMsgSystemVPCNetCfgNotFound.getNSNetworkCondition(err))
 		return common.ResultRequeueAfter10sec, err
 	}
 
+	retryWithSystemVPC := false
+	var systemNSCondition *corev1.NamespaceCondition
+
+	gatewayConnectionReady, _ := getGatewayConnectionStatus(ctx, systemVpcNetCfg)
 	gatewayConnectionReason := ""
 	if !gatewayConnectionReady {
-		if ncName == commonservice.SystemVPCNetworkConfigurationName {
-			gatewayConnectionReady, gatewayConnectionReason, err = r.Service.ValidateGatewayConnectionStatus(&nc)
-			log.Info("got the gateway connection status", "gatewayConnectionReady", gatewayConnectionReady, "gatewayConnectionReason", gatewayConnectionReason)
-			if err != nil {
-				log.Error(err, "Failed to validate the edge and gateway connection", "Org", nc.Org, "Project", nc.NSXProject)
-				updateFail(r, ctx, networkInfoCR, &err, r.Client, nil)
-				return common.ResultRequeueAfter10sec, err
-			}
-			setVPCNetworkConfigurationStatusWithGatewayConnection(ctx, r.Client, vpcNetworkConfiguration, gatewayConnectionReady, gatewayConnectionReason)
-		} else {
+		// Retry after 60s if the gateway connection is not ready in system VPC.
+		if ncName != commonservice.SystemVPCNetworkConfigurationName {
 			log.Info("Skipping reconciliation due to unready system gateway connection", "NetworkInfo", req.NamespacedName)
+			setNSNetworkReadyCondition(ctx, r.Client, req.Namespace, nsMsgVPCGwConnectionNotReady.getNSNetworkCondition())
 			return common.ResultRequeueAfter60sec, nil
 		}
+
+		// Re-check the gateway connection readiness in system VPC on NSX.
+		gatewayConnectionReady, gatewayConnectionReason, err = r.Service.ValidateGatewayConnectionStatus(&nc)
+		log.Info("got the gateway connection status", "gatewayConnectionReady", gatewayConnectionReady, "gatewayConnectionReason", gatewayConnectionReason)
```

Comment:
got --> Got

Reply:
Using lower case in log because @TaoZou1 used to comment (https://github.com/vmware-tanzu/nsx-operator/pull/799#discussion_r1808119765) that it is required in nsx-operator repo, also confirmed by Danting and Sean. So I would leave it as it was now.